### ISSUE TITLE
folder_contents broken on site upgraded from 4.x

### ISSUE THREAD

**Issue by fulv** [plone/Products.CMFPlone#1283]
I upgraded a site from 4.0.10 to 5.0, and when I go to folder_contents, I get the whole UI but nothing listed below the table column headers.  There is a spinner, and this traceback:

```
2015-12-07 23:07:19 ERROR Zope.SiteErrorLog 1449558439.230.227105431097 http://localhost:8080/Plone/@@getVocabulary
Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module ZPublisher.Publish, line 48, in call_object
  Module plone.app.content.browser.vocabulary, line 76, in __call__
  Module plone.app.content.browser.vocabulary, line 230, in get_vocabulary
  Module plone.app.vocabularies.catalog, line 540, in __call__
  Module plone.app.querystring.queryparser, line 36, in parseFormquery
  Module zope.dottedname.resolve, line 38, in resolve
ImportError: No module named _path
2015-12-07 23:07:19 ERROR Zope.SiteErrorLog 1449558439.230.370313065892 http://localhost:8080/Plone/@@getVocabulary
Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module ZPublisher.Publish, line 48, in call_object
  Module plone.app.content.browser.vocabulary, line 76, in __call__
  Module plone.app.content.browser.vocabulary, line 230, in get_vocabulary
  Module plone.app.vocabularies.catalog, line 540, in __call__
  Module plone.app.querystring.queryparser, line 36, in parseFormquery
  Module zope.dottedname.resolve, line 38, in resolve
ImportError: No module named _path
```

**Comment by fulv** [comment]
Actually, there is very easy way to reproduce this:

  1. Create a new site in Plone 4.3.7
  2. Install plone.app.theming at a minimum, and perhaps also plone.app.contenttypes so you don't have to migrate later
  3. Move Data.fs and blobstorage to Plone 5.0 var 
  4. Start 5.0, and upgrade
  5. Go to Theming control panel to activate barceloneta
  6. go home and click on Contents

**Comment by pbauer** [comment]
@fulv try if running the complete profile of plone.app.querystring resolves this. I guess there is missing upgrade-step in the changes that @rnixx did to ``_path``. As a fallback you could simply add a bbb-method ``_path`` that calls ``_navigationPath`` in ``plone.app.querystring.queryparser``.

**Comment by hvelarde** [comment]
@vangheem seems to be related with #1306

**Comment by fulv** [comment]
Since this is one of those cases where the name of the product is not actually visible in the UI (same as `plone.app.iterate`) and normal mortals like me would never figure out what _running the complete profile of plone.app.querystring_ actually means, I'm going to spell out exactly what I did:

  - go to ZMI, `portal_setup`
  - Select Profile `Querystring parser/builder`
  - at the bottom, under **Import all steps**, leave the first radio button selected (i.e. _Apply new profiles_)
  - click _Import all steps_

After this, folder_contents works fine.

**Comment by pbauer** [comment]
The upgrade-step upgrade_to_5 of plone.app.querystring is never run. We need to run it in plone.app.upgrade explicitly like this when upgrading from 5.0.1 to 5.0.2 (it is too late now for 5.0.1):

In the upgrade-step from Plone 5.0b4 to 5.0b5 the profile of plone.app.querystring is updated but it's profile-version is already at 5 at that point so the step upgrade_to_5 is not run. 

pr is here: https://github.com/plone/plone.app.upgrade/commit/3bcedea5f351a58b4f080918b28f4cd5b5398343

**Comment by gforcada** [comment]
@fulv given @pbauer answer and that p.a.upgrade pull request is already released on 5.0.2 I'm closing this one, reopen if it's still happening.
